### PR TITLE
[IOTDB-2461] Add cpp and sql support for ZIGZAG encoding

### PR DESCRIPTION
**Pull request by HTHou**
## Description

Additional code of PR #4939 
Add cpp and sql support for ZIGZAG encoding.

### CODE REVIEW

File: integration/src/test/java/org/apache/iotdb/db/integration/IoTDBEncodingIT.java [hunk 1]
```diff
@@ -243,6 +243,62 @@ public void testSetTimeEncoderRegularAndValueEncoderGORILLAOutofOrder() {
...
+      try (ResultSet resultSet = statement.executeQuery("select * from root.db_0.tab0")) {
+        int index = 0;
+        while (resultSet.next()) {
+          int salary = resultSet.getInt("root.db_0.tab0.salary");
+          assertEquals(result[index], salary);
+          index++;
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
```
Comment (qiaojialin) [inline integration/src/test/java/org/apache/iotdb/db/integration/IoTDBEncodingIT.java:270]: add a fail();

File: integration/src/test/java/org/apache/iotdb/db/integration/IoTDBEncodingIT.java [hunk 2]
```diff
@@ -243,6 +243,62 @@ public void testSetTimeEncoderRegularAndValueEncoderGORILLAOutofOrder() {
...
```
Comment (qiaojialin) [inline integration/src/test/java/org/apache/iotdb/db/integration/IoTDBEncodingIT.java:298]: add a fail();

File: integration/src/test/java/org/apache/iotdb/db/integration/IoTDBEncodingIT.java (hunk 1 above)
Comment (HTHou) [reply integration/src/test/java/org/apache/iotdb/db/integration/IoTDBEncodingIT.java:277]: Fixed

File: integration/src/test/java/org/apache/iotdb/db/integration/IoTDBEncodingIT.java (hunk 2 above)
Comment (HTHou) [reply integration/src/test/java/org/apache/iotdb/db/integration/IoTDBEncodingIT.java:306]: Fixed

**Review by wangchao316 — APPROVED** [review]
LGTM

**Pull request merged**